src/test/java/unit/test/JsonParserTest.java

```diff
@@ -0,0 +1,88 @@
+package unit.test;
+
+import org.json.JSONException;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import parser.Parser;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JsonParserTest {
+
+    private Parser parser;
+    private Cart cart;
+    private File file;
+
+    @BeforeEach
+    void setUp() {
+        file = new File("src/main/resources/test-cart2.json");
+        cart = new Cart("test-cart2");
+
+        RealItem realItem = new RealItem();
+        realItem.setName("Volvo");
+        realItem.setPrice(1520);
+        realItem.setWeight(1200);
+
+        VirtualItem virtualItem = new VirtualItem();
+        virtualItem.setName("Vista");
+        virtualItem.setPrice(1600);
+        virtualItem.setSizeOnDisk(200);
+
+        cart.addRealItem(realItem);
+        cart.addVirtualItem(virtualItem);
+
+        parser = new JsonParser();
+    }
+
+    @Test
+    void checkFileExists() {
+        parser.writeToFile(cart);
+        assertTrue(file.exists());
```

Comment:
Always add second parameter (message on failure) for assertTrue/assertFalse, otherwise you will spend time to understand what is incorrect/where is bug.